### ISSUE TITLE
Mojom validation tests: fix validation error code inconsistencies

### ISSUE THREAD

**Issue by vardhan-cr** [domokit/mojo#811]
struct header validation returns ILLEGAL_MEMORY_RANGE, as opposed to UNEXPECTED_STRUCT_HEADER.  if the struct header's size is larger than the payload itserlf.  for arrays, validation returns UNEXPECTED_ARRAY_HEADER.

review any other inconsistencies.  fixing this should involve touching bindings for all supported languages.

**Comment by j9brown** [comment]
Out if curiosity, how important is it to distinguish all of these cases?
(Would the code be more compact / faster if it only reported success /
failure?)

**Comment by mknyszek** [comment]
I recently found myself dealing with this case. The data file that expects ILLEGAL_MEMORY_RANGE as the error just contains "0x00" which is different from the other data file which expects UNEXPECTED_STRUCT_HEADER (but also has size 0). This led me to believe the error that was being checked here was not that the struct header size was broken, but that the byte buffer produced by the validation test parser isn't even large enough to contain a struct header in the first place, hence the "illegal memory range". This may be wrong (especially because on some level the error really is a broken struct header) but it made sense to me at the time.

**Comment by vardhan-cr** [comment]
@j9brown I don't think we save save /that/ much by getting rid of the error codes (maybe just code size).  if you receive invalid data from a trusted source, you can use the codes to figure out whats wrong, though I dont think they're useful enough at the moment, to be honest (esp. considering their inconsistencies, and lack of additional error strings).  OTOH if you're receive invalid data from an untrusted source, there's probably no point in figuring out what's wrong with the data

**Comment by j9brown** [comment]
It's mostly a problem of what people are going to do with the information.
I can't envision writing a program which discriminated between these fine
grained validation errors.  If they did, it might also introduce the
potential for compatibility problems down the road if we ever change how
the validator functions.

**Comment by vardhan-cr** [comment]
Right -- there isn't a programmatic thing you can do depending on the error, if I'm interpreting your comment correcting.  The validation error codes aren't very useful outside of debugging, and currently not useful enough because of how many different cases each error code covers, so we need to differentiate them.

I think differentiating them (either via the current error codes or replacing them with more human-friendly error strings) is very useful for debugging though.
